pkg/db/migration.go

```diff
@@ -228,3 +236,109 @@ func MigrateVersion(ctx context.Context, dbPool Database, params params.Database
 	}
 	return version, dirty, err
 }
+
+// TODO: (niro) Remove on feature complete
+
+// kvMigrate implementation of KV Migration logic
+func kvMigrate(dbParams params.Database, m *migrate.Migrate, drop bool) error {
+	if !dbParams.KVEnabled {
+		return nil
+	}
+	ctx := context.Background()
+	dbPool := BuildDatabaseConnection(ctx, dbParams)
+	defer dbPool.Close()
+	version, dirty, err := MigrateVersion(ctx, dbPool, dbParams)
+	if err != nil {
+		return err
+	}
+	if dirty {
+		return migrate.ErrDirty{Version: int(version)}
+	}
+	if kv.MigrateVersion != version {
+		return migrate.ErrInvalidVersion
+	}
+
+	// Delete store if exists from previous failed KV migration
+	err = dropTables(ctx, dbPool, []string{kvpg.DefaultTableName})
+	if err != nil {
+		return err
+	}
+
+	kvStore, err := kv.Open(ctx, kvpg.DriverName, dbParams.ConnectionString)
+	if err != nil {
+		return err
+	}
+
+	// Import to KV Store
+	var g multierror.Group
+	for n, f := range kvPkgs {
+		name := n
+		migrateFunc := f
+		g.Go(func() error {
+			logging.Default().Info("Starting KV migration for package:", name)
+			buf := bytes.Buffer{}
+			err := migrateFunc(dbPool, &buf)
+			if err != nil {
+				logging.Default().WithError(err).Error("Failed migration on pkg:", name)
+				return err
+			}
+			err = export.ImportFile(ctx, &buf, kvStore)
+			if err != nil {
+				logging.Default().WithError(err).Error("Failed export on pkg:", name)
+				return err
+			}
+			logging.Default().Info("Successfully migrated pkg", name, "to KV")
+			return nil
+		})
+	}
+	err = g.Wait().ErrorOrNil()
+	if err != nil {
+		return err
+	}
+
+	// Update migrate version
+	err = m.Force(kv.MigrateVersion + 1)
```

Comment:
I prefer not to bring the migration to invalid state that that we will need to maintain a constant in order to verify.
If we assume that the latest migration should always be applied the 'kv' store should hold the information if the data-migration was completed or not.